vllm/model_executor/models/paligemma.py

```diff
@@ -149,7 +154,10 @@ def __init__(self, *, vllm_config: VllmConfig, prefix: str = ""):
             projection_dim=config.vision_config.projection_dim)
 
         self.quant_config = quant_config
-        config.text_config.architectures = ["GemmaForCausalLM"]
+
+        # PaliGemma 2 configs have the architecture set to Gemma2ForCausalLM
+        if config.text_config.architectures is None:
+            config.text_config.architectures = ["GemmaForCausalLM"]
```

Comment:
I prefer to update this based on `model_type`, because both paligemma and paligemma2 have explicit `model_type` in text config.

Reply:
Updated.